libenv/sysinfo.c

```diff
@@ -1645,6 +1659,88 @@ static void OSClasses(EvalContext *ctx)
 
 /*********************************************************************************/
 
+#ifdef __APPLE__
+static void Apple_Version(EvalContext *ctx)
+{
+    FILE *pp = NULL;
+
+    if ((!FileCanOpen("/usr/bin/sw_vers", "r") || ((pp = cf_popen("/usr/bin/sw_vers", "r", true)) == NULL)))
+    {
+        Log(LOG_LEVEL_VERBOSE, "Could not find macOS system version information.");
+        return;
+    }
+
+    size_t line_size = CF_BUFSIZE;
+    char *line = xmalloc(line_size), *r;
+    int revcomps = 0, major, minor, patch;
+    char *flavor = NULL;
+
+    while (CfReadLine(&line, &line_size, pp) != -1)
+    {
+        if ((r = strstr(line, "ProductName")))
+        {
+            r = strrchr(line, '\t');
+            if (!r || !++r)
```

Comment:
Please use explicit comparisons as described in [CONTRIBUTING.md](https://github.com/cfengine/core/blob/master/CONTRIBUTING.md)
```suggestion
            if (r == NULL || ++r == NULL)
```